### ISSUE TITLE
PyAutoFEP example failed

### ISSUE THREAD

**Issue by yx0516** [luancarvalhomartins/PyAutoFEP#19]
### Discussed in https://github.com/luancarvalhomartins/PyAutoFEP/discussions/18

<div type='discussions-op-text'>

<sup>Originally posted by **luancarvalhomartins** November 12, 2021</sup>
### PyAutoFEP discussions are now active
Thanks you very much for your interest in PyAutoFEP

From the activity in issues and from emails I get every once and a while, I feel that there is a need of a better place for users to ask questions and share ideas for PyAutoFEP.

Also, I plan to use this space to share the roadmap and ongoing updates get feedback about it. I hope I can do some kind of weekly/biweekly post of what I have been coding, especially about the GUI.</div>

**Comment by yx0516** [comment]
I encountered this problem during the reproduction of the pyautoFEP example. The corresponding processing steps in the example were executed without errors, and finally the command was submitted and executed for some time (several hours) and then the following error appear
![_16370207702660](https://user-images.githubusercontent.com/13602182/141881934-aa2d2b93-e813-4814-85fe-5e3f643580b8.png)
ed

**Comment by baba-hashimoto** [comment]
* will be the name of the folder, such as FXR_12-FXR_74 or FXR_12-FXR_76. Access that folder and see if the data is there.

**Comment by luancarvalhomartins** [comment]
This is the output of tar telling you that the files were not found. As @baba-hashimoto said, take a look at `FXR_12-FXR_XX/protein/md`, specially in the `md/lambdaX` directories. It seems that your MD didn't run or didn't complete.

I have a long term plan to rewrite the run scripts, as currently the log is not very useful.

**Comment by xuyijie100** [comment]
I also encountered this problem . My program ran min, nve, nvt, npt normally, but skipped md. I really need your help and thank you very much!

**Comment by luancarvalhomartins** [comment]
This is likely caused by using a GROMACS binary that was not compiled for MPI support. Make sure your GROMACS binary in `gmx_bin_run` supports MPI (is, it points to your `gmx_mpi`). Would you mind checking this?

Otherwise, would you mind sending the error message from the log file in the perturbation run dir (eg, in `FXR_12-FXR_74/water/md`)?

**Comment by xuyijie100** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by GsGithub17** [comment]
Hi all,

I have the same problem. See attached screenshot.

What was the solution for this?

Thanks! 

![EFPerror](https://user-images.githubusercontent.com/29179511/161832306-5c37ba8e-90e3-42cd-a827-f08d23b0b928.PNG)

**Comment by xuyijie100** [comment]
I reinstalled GROMACS with MPI, also you should make sure that the CPU has 12 or more cores when running, otherwise you need to complete the simulation through hyper threading.

**Comment by GsGithub17** [comment]
I run it with GROMACS enabled with MPI, how every I can't understand what the requirement of 12core/CPU is based on.
Thanks!

**Comment by luancarvalhomartins** [comment]
Sorry for the tardiness. Replying in a closed issue confused me a bit. Opening a new discussion (or even a new issue) probably works better.

It's not that 12 cores are required, you can run using less than that. `gmx_mpi` will spawn a MPI-process per replica, so that N gmx processes will be run in parallel, communicating by MPI (N=12 by default). If you run in a node with less than N cores, GROMACS will oversubscribe processes to cores, so that some of the replicas will be forced to share cores. This will likely hurt your performance badly, so gmx_mpi will warn you about that. I sometimes use 8-core desktops with gaming GPUs for testing and they are able to run 12 or even 24 replicas, but **much** slower than a high-performance cluster node.

Further reading:
 - https://manual.gromacs.org/documentation/current/user-guide/mdrun-features.html#running-multi-simulations
 - https://www.gromacs.org/Documentation/Acceleration_and_parallelization
 - https://manual.gromacs.org/documentation/current/user-guide/mdrun-performance.html

**Comment by zirui** [comment]
I'm experiencing the same issue,  and i could not find any hit in the logs.
so i run step one by one in `runall.sh`,  and get error after run 'bash  FXR_12-FXR_74/runall_FXR_12-FXR_74_protein.sh', 

error logs in  `FXR_12-FXR_74/protein/lambda0/min01/min01_grompp.log`

```
Number of degrees of freedom in T-Coupling group rest is 70821.00
ERROR 1 [file FXR_12-FXR_74/protein/lambda0/min01/min01.mdp]:
  The largest distance between excluded atoms is 1.063 nm, which is larger
  than the cut-off distance. This will lead to missing long-range
  corrections in the forces and energies.
```

**Comment by baba-hashimoto** [comment]
@zirui 
Which version of Gromacs do you use?
Gromacs 2022 has improved detection of errors or theoretically faulty settings.
As a result, files used for sample calculations do not work.

The settings stored in templates/mdp/default_nosc/water need to be changed.
The specific changes are as follows.
; Electrostatics.
coulombtype = PME
pme-order = 4
fourierspacing = 0.16
rvdw = 1.2
rlist = 1.2
rcoulomb = 1.2

The values set in protein would also need to be adjusted to our settings.

 @luancarvalhomartins
With Gromacs2022, the default samples no longer work. If you have time to check, we would appreciate it.

**Comment by luancarvalhomartins** [comment]
@baba-hashimoto Thanks for letting me know that. I wasn't able to test the 2022 version yet - quite swamped here. Do you have more info on why the defaults for `rvdw`, `rlist` and `rcoulomb` fails (your values for `coulombtype`, `pme-order`, and `fourierspacing` already matches the ones in **templates/mdp/default_nosc/water/*.mpd**)?

Also, does 2022 uses `rlist` and `rcoulomb`? From the manual, it seems that these values should be optimized on the fly (https://manual.gromacs.org/documentation/current/user-guide/mdp-options.html#mdp-rlist and https://manual.gromacs.org/documentation/current/user-guide/mdp-options.html#mdp-rcoulomb)? 

Thank you very much

**Comment by zirui** [comment]
> @zirui Which version of Gromacs do you use? Gromacs 2022 has improved detection of errors or theoretically faulty settings. As a result, files used for sample calculations do not work.
> 
> The settings stored in templates/mdp/default_nosc/water need to be changed. The specific changes are as follows. ; Electrostatics. coulombtype = PME pme-order = 4 fourierspacing = 0.16 rvdw = 1.2 rlist = 1.2 rcoulomb = 1.2
> 
> The values set in protein would also need to be adjusted to our settings.

thanks for you reply @baba-hashimoto ,my Gromacs version is 2022.1.
i changed the settings as you suggested,  but failed in the end

```
+ bash pack.sh
tar: */water/md/lambda*/lambda_pbcfit.xtc: Cannot stat: No such file or directory
tar: */water/md/lambda*/analysis.log: Cannot stat: No such file or directory
tar: */water/md/analysis/*.xvg: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
```

**Comment by luancarvalhomartins** [comment]
@zirui That final error is just tar telling you it didn't find some of the output files for water perturbations. You can safely ignore those, for now.

**Comment by zirui** [comment]
I ignored the error above, and try to run post analysis,
```
python ../../../analyze_results.py --input tutorial.tgz --units kcal --output_uncompress_directory tutorial --center_molecule=FXR_12
```
and failed:
```
[ERROR] Could not find unk_matrix.pkl file in tutorial/FXR_12-FXR_76/protein/md/rerun to read temperature from. Also, could not guess temperature from MD log in tutorial/FXR_12-FXR_76/protein/md/rerun/../lambda0/lambda.log. Set temperature argument or config option. Analysis for this system will not be run.
=================== STACK INFO ===================
  File "../../../analyze_results.py", line 1511, in <module>
    verbosity=arguments.verbose)
  File "../../../analyze_results.py", line 1173, in collect_from_xvg
    msg_verbosity=os_util.verbosity_level.error, current_verbosity=verbosity)
  File "/home/PyAutoFEP/os_util.py", line 292, in local_print
    formatted_string += '\n{:=^50}\n{}{:=^50}'.format(' STACK INFO ', ''.join(traceback.format_stack()),
=================== STACK INFO ===================
[ERROR] Could not find unk_matrix.pkl file in tutorial/FXR_12-FXR_85/protein/md/rerun to read temperature from. Also, could not guess temperature from MD log in tutorial/FXR_12-FXR_85/protein/md/rerun/../lambda0/lambda.log. Set temperature argument or config option. Analysis for this system will not be run.
=================== STACK INFO ===================
  File "../../../analyze_results.py", line 1511, in <module>
    verbosity=arguments.verbose)
  File "../../../analyze_results.py", line 1173, in collect_from_xvg
    msg_verbosity=os_util.verbosity_level.error, current_verbosity=verbosity)
  File "/home/PyAutoFEP/os_util.py", line 292, in local_print
    formatted_string += '\n{:=^50}\n{}{:=^50}'.format(' STACK INFO ', ''.join(traceback.format_stack()),
=================== STACK INFO ===================
================== Pairwise ΔΔG ==================
         Perturbation           protein    water
Traceback (most recent call last):
  File "../../../analyze_results.py", line 1661, in <module>
    perturbation_data = get_data_from_unk_pkl(files['pkl'])
  File "../../../analyze_results.py", line 1197, in get_data_from_unk_pkl
    with open(input_pkl, 'rb') as fh:
FileNotFoundError: [Errno 2] No such file or directory: 'tutorial/FXR_12-FXR_76/protein/md/rerun/unk_matrix.pkl
```
it seems that  the expected file  `unk_matrix.pkl` was not generated for `FXR_12-FXR_76/protein` and `FXR_12-FXR_85/protein` in FEP step.

all `unk_matrix.pkl` files in tutorial directory:
```
./tutorial/FXR_12-FXR_88/protein/md/rerun/unk_matrix.pkl
./tutorial/FXR_12-FXR_88/water/md/rerun/unk_matrix.pkl
./tutorial/FXR_12-FXR_76/water/md/rerun/unk_matrix.pkl
./tutorial/FXR_12-FXR_74/protein/md/rerun/unk_matrix.pkl
./tutorial/FXR_12-FXR_74/water/md/rerun/unk_matrix.pkl
./tutorial/FXR_12-FXR_84/protein/md/rerun/unk_matrix.pkl
./tutorial/FXR_12-FXR_84/water/md/rerun/unk_matrix.pkl
./tutorial/FXR_12-FXR_85/water/md/rerun/unk_matrix.pkl
```

**Comment by luancarvalhomartins** [comment]
The `unk_matrix.pkl` files for the FXR_12-FXR_76 protein leg and FXR_12-FXR_85 protein leg are not there. `analyze_results.py` will recreate such files if the `.xvg` outputs are present. From your output, it seems that these legs were not run. Please, check the outputs from these legs to see what happened.

**Comment by xuyijie100** [comment]
Thanks for your help in advance.&nbsp; The log file named mdout.log in FXR_12-FXR_14/water/lambda0/md is provided here. Thank you again for giving us an easy way to use fep. Feel free to contact me if there is anything I can help with. 





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "luancarvalhomartins/PyAutoFEP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月18日(星期五) 凌晨0:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [luancarvalhomartins/PyAutoFEP] PyAutoFEP example failed (Issue #19)





 
This is likely caused by using a GROMACS binary that was not compiled for MPI support. Make sure your GROMACS binary in gmx_bin_run supports MPI (is, it points to your gmx_mpi). Would you mind checking this?
 
Otherwise, would you mind sending the error message from the log file in the perturbation run dir (eg, in FXR_12-FXR_74/water/md)?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Layne-Huang** [comment]
> This is likely caused by using a GROMACS binary that was not compiled for MPI support. Make sure your GROMACS binary in `gmx_bin_run` supports MPI (is, it points to your `gmx_mpi`). Would you mind checking this?
> 
> Otherwise, would you mind sending the error message from the log file in the perturbation run dir (eg, in `FXR_12-FXR_74/water/md`)?

I did not run the nve and nvt successfully. And it throw out the errors:

`7l11_ligand-7l11_3746/runall_7l11_ligand-7l11_3746_protein.sh: line 14: 114419 Segmentation fault      (core dumped) gmx_mpi mdrun --deffnm nve > nve_mdrun_log.log 2>&1
7l11_ligand-7l11_981/runall_7l11_ligand-7l11_981_protein.sh: line 33: 116577 Segmentation fault      (core dumped) gmx_mpi mdrun --deffnm nvt > nvt_mdrun_log.log 2>&1
tar: */water/md/rerun/*.pkl: Cannot stat: No such file or directory
tar: */water/md/rerun/*.xvg: Cannot stat: No such file or directory
tar: */water/md/rerun/*.edr: Cannot stat: No such file or directory
tar: */protein/md/rerun/*.pkl: Cannot stat: No such file or directory
tar: */protein/md/rerun/*.xvg: Cannot stat: No such file or directory
tar: */protein/md/rerun/*.edr: Cannot stat: No such file or directory
tar: */protein/md/lambda*/lambda_pbcfit.xtc: Cannot stat: No such file or directory
tar: */protein/md/lambda*/lambda.log: Cannot stat: No such file or directory
tar: */protein/md/lambda*/lambda.edr: Cannot stat: No such file or directory
tar: */water/md/lambda*/lambda.xtc: Cannot stat: No such file or directory
tar: */water/md/lambda*/lambda_nocont.tpr: Cannot stat: No such file or directory
tar: */water/md/lambda*/lambda.tpr: Cannot stat: No such file or directory
tar: */water/md/lambda*/lambda.log: Cannot stat: No such file or directory
tar: */water/md/lambda*/lambda.edr: Cannot stat: No such file or directory
tar: */protein/md/lambda*/analysis.log: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors`

Could please tell me how to solve it?
